packages/ui/components/EditSelfPasswordModal/EditSelfPasswordModal.tsx

```diff
@@ -0,0 +1,128 @@
+import { FunctionComponent, useState, useEffect } from 'react'
+import { ButtonPopup, Button, TextInput } from '../..'
+import { btnType } from '../Button/Button'
+import PasswordMeter from '../PasswordMeter/PasswordMeter'
+import zxcvbn from 'zxcvbn'
+
+type Props = {
+  onSubmit: (currentPassword: string, newPassword: string) => void
+}
+
+export const EditSelfPasswordModal: FunctionComponent<React.PropsWithChildren<Props>> = ({ onSubmit }) => {
+  const [oldPassword, handleOldPassword] = useState<string>('')
+  const [newPassword, handleNewPassword] = useState<string>('')
+  const [confirmPassword, handleConfirmPassword] = useState<string>('')
+  const [showValidations, handleShowValidations] = useState<boolean>(false)
+  const [passwordStrength, handlePasswordStrength] = useState<number>(0)
+
+  useEffect(() => {
+    handlePasswordStrength(zxcvbn(newPassword).score)
+  }, [newPassword])
+
+  return (
+    <ButtonPopup       
+      toggleButton={(toggle) => (
+        <Button
+          text="CHANGE PASSWORD"
+          onClick={toggle}
+          type={btnType.Secondary}
+        />
+      )}
+      render={(toggle) => (
+        <div className='p-4'>
+          <p className='py-2 font-sans text-gray-600 text-xxxl font-bold'>
+            Edit password
+          </p>
+          <p className='font-sans text-sm font-normal text-gray-500'>
+            Please enter your current password and new password.           
+          </p>
+          
+          <div className='py-4'>
+            <TextInput
+              name='current-password'
+              placeholder='Current password'
+              type='password'
+              value={oldPassword}
+              onChange={(e) => { handleOldPassword(e.target.value) }}
+              onFocus={(e) => {
+                if (!e.currentTarget.contains(e.relatedTarget)) {
```

Comment:
can we create a reusable function for both onFocus and onBlur?

Reply:
So I noticed is not exactly the same function but we could adjust this way:

```
const handleInput = (e: FocusEvent, isFocus: boolean) => {
  if (!e.currentTarget.contains(e.relatedTarget)) {
    handleShowValidations(!isFocus)
  }
}
```

and then call

onFocus={(e) => handleInput(e, true)}
onBlur={(e) => handleInput(e, false)}